From 4890063cf90cb54ba2db7904404ad5e57a56c1cf Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
Date: Wed, 1 May 2019 03:00:12 +0900
Subject: [PATCH 4/5] spp_pcap: add lcore IDs in status message

Add lcore IDs in response of status command to know which of lcores are
in use.

Signed-off-by: Yasufumi Ogawa <ogawa.yasufumi@lab.ntt.co.jp>
---
 src/pcap/command_proc.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/pcap/command_proc.c b/src/pcap/command_proc.c
index 28c6b97..96218c0 100644
--- a/src/pcap/command_proc.c
+++ b/src/pcap/command_proc.c
@@ -523,6 +523,16 @@ append_pcap_core_element_value(
 	return ret;
 }
 
+/* append master lcore in JSON format */
+static int
+append_master_lcore_value(const char *name, char **output,
+		void *tmp __attribute__ ((unused)))
+{
+	int ret = SPP_RET_NG;
+	ret = append_json_int_value(name, output, rte_get_master_lcore());
+	return ret;
+}
+
 /* append a list of core information for JSON format */
 static int
 append_core_value(const char *name, char **output,
@@ -625,6 +635,7 @@ struct command_response_list response_result_list[] = {
 struct command_response_list response_info_list[] = {
 	{ "client-id",        append_client_id_value },
 	{ "status",           append_capture_status_value },
+	{ "master-lcore",     append_master_lcore_value },
 	{ "core",             append_core_value },
 	COMMAND_RESP_TAG_LIST_EMPTY
 };
-- 
2.17.1

